internal/webhook/userinfo.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	ServiceAccountEnvName = "SERVICE_ACCOUNT"
+)
+
+func serviceAccountIsEqual(req admission.Request, serviceAccount string) bool {
+	return strings.HasPrefix(req.UserInfo.Username, "system:serviceaccount:") &&
```

Comment:
I think you can use `SplitUsername` from "k8s.io/apiserver/pkg/authentication/serviceaccount" here